1.18_linux/runtime/cgo/gcc_libinit.c.patch

```diff
@@ -5,3 +5,24 @@
   #define _GNU_SOURCE
 #endif
 #include <pthread.h>
+//--from
+		err = pthread_create(thread, attr, pfn, arg);
+		if (err == 0) {
+			pthread_detach(*thread);
+			return 0;
+		}
+//--to
+		err = pthread_create(thread, attr, pfn, arg);
+		if (err == 0) {
+			pthread_detach(*thread);
+#if __aarch64__
```

Comment:
This is a way of saying we only want this when targeting the Switch. The PlayStation pthread API does not have `pthread_setaffinity_np`. Should I add a comment maybe?

Reply:
Then, can we have a better way to detect whether `pthread_setaffinity_np` is defined or not?

Reply:
Also, in PlayStation, don't we need to specify affinities for a thread?

Reply:
Well everything seems to work without specifying affinities on the PlayStation.

But now I realise that if the user doesn’t set `cfg.numCPU` it defaults to the build machine’s settings… that’s why I had no problems on the console, since it has as many CPUs as my desktop… I think I will need to redo this PR in another way.

Reply:
How about removing `cfg.numCPU` and one of the following:
 - let the user override `c_sched_getaffinity()` with their own implementation
 - provide a new `get_cpu_count()` that is used instead of `HITSUMABUSHI_NUMCPU` and that the user can also override

Most targets, including the Switch, are able to provide their number of CPUs/cores. I think this would be more future-proof than `cfg.numCPU`.

Reply:
> let the user override c_sched_getaffinity() with their own implementation

This implementation should be determined based on the number of CPUs, right? Do we really allow overwriting this?

Reply:
Well I think the user is best suited to know how many CPUs there are on the platform. There are Sony and Nintendo APIs to retrieve this information at runtime (or even `std::thread::hardware_concurrency();` in some cases). I like this strategy better than hardcoding how many cores a given console has, and a lot more than defaulting to the build machine’s runtime value…

Reply:
I'm fine to make something overwritten by a user for a number of CPUs, but I doubt we really need to allow overwriting c_sched_getaffinity.

Reply:
That was my suggestion because the function is already there and `sched_getaffinity()` is what Go uses to define the number of CPUs (`getproccount()` in `os_linux.go`), but I’m fine with introducing a new function. What would be a good name? `hitsumabushi_getproccount()` maybe?

Reply:
I see. `hitsumabushi_getproccount()` sounds fine.